modules/claims_api/spec/requests/v2/veterans/rswag_power_of_attorney_spec.rb

```diff
@@ -1071,7 +1071,7 @@
               'application/json' => {
                 example: JSON.parse(response.body, symbolize_names: true).tap do |json|
                   json[:data][:attributes]
-                    .merge!(procId: '3857415')
+                    .merge!(id: '12e13134-7229-4e44-90ae-bcea2a4525fa', type: 'power-of-attorney-request')
```

Comment:
a little caught off guard on why we have to merge these in since they should be part of the request response?

Reply:
Looking at the workflow if we handle some of the before stuff to match the workflow more (I believe ?) we should be able to avoid just stapling these on at the end

```
            allow_any_instance_of(ClaimsApi::PowerOfAttorneyRequestService::Orchestrator)
              .to receive(:submit_request)
              .and_return({'procId' => '12345'})

            allow(Flipper).to receive(:enabled?).with(:lighthouse_claims_v2_poa_requests_skip_bgs).and_return(false)
```

Let me know if I am just completely missing something obvious here, docs look fine, just caught off guard by seeing this `.merge` adding the values when they are on the response

Reply:
<img width="81" alt="passing" src="https://github.com/user-attachments/assets/7160e998-957d-4826-b2d5-b677bb1774c4" />

woohoo!

Reply:
@tycol7  Unfortunately I think even though that change passes green, it will not generate the ID and Type in the docs as expected and so the next person that runs them will erase this update like in my compile below

![Screenshot 2024-12-11 at 4 13 53 PM](https://github.com/user-attachments/assets/f81de34b-1663-4848-82e4-078bd86d83ad)

 I think in order to get what you are looking for a few things need to be added.  I tried to do a 'suggestion' on this but it does not let me do that on lines that are not changed so I will just give line numbers.

- need to return a `procId` in the `submit_request` so on line 1058 change the return to `.and_return({'procId' => '12345'})` (or whatever number you want)
- Disable the flipper in the `before do` block as well anywhere lines 1055 -1063 `allow(Flipper).to receive(:enabled?).with(:lighthouse_claims_v2_poa_requests_skip_bgs).and_return(false)`

Then when the docs compile it will be able to include those values like my next compile below

![Screenshot 2024-12-11 at 4 13 53 PM](https://github.com/user-attachments/assets/0a102d53-1ec5-461a-bcf6-f98a11f4e4c3)

Reply:
ah, right! changes here: https://github.com/department-of-veterans-affairs/vets-api/pull/19756/commits/9b461ee03ebffb1b1c87a14752555854844a9a16. ran the doc generator and test with expected results.